[doc][core][dashboard][state] Promote state api and dashboard usage in core user guide. (#35760)
ray-project/ray: doc/source/ray-core/actors/terminating-actors.rst

flag ``no_restart=False`` to ``ray.kill``.

For :ref:`named and detached actors <actor-lifetimes>`, calling ``ray.kill`` on
an actor handle destroys the actor and allow the name to be reused.

> From Ray 2.0 onwards, you should be able to view the death cause of an actor in `death_cause` as part of the actor state by querying for information about the actor, with the :ref:`State API <state-api-overview-ref>`, or see the Actors View in the :ref:`Dashboard <observability-getting-started>`.

Reviewer: "use `ray list actors --detail <link to the page>` to see a list of dead actors and the reason".

```
# This API is only available when Ray is downloaded by `pip install "ray[default]"`
ray list actors
```

```
  ---
  -   actor_id: e8702085880657b355bf7ef001000000
      class_name: Actor
      state: DEAD
      job_id: '01000000'
      name: ''
      node_id: null
      pid: 0
      ray_namespace: dbab546b-7ce5-4cbb-96f1-d0f64588ae60
      serialized_runtime_env: '{}'
      required_resources: {}
      death_cause:
          actor_died_error_context: # <---- You could see the error message w.r.t why the actor exits. 
              error_message: The actor is dead because `ray.kill` killed it.
              owner_id: 01000000ffffffffffffffffffffffffffffffffffffffffffffffff
              owner_ip_address: 127.0.0.1
              ray_namespace: dbab546b-7ce5-4cbb-96f1-d0f64588ae60
              class_name: Actor
              actor_id: e8702085880657b355bf7ef001000000
              never_started: true
              node_ip_address: ''
              pid: 0
              name: ''
      is_detached: false
      placement_group_id: null
      repr_name: ''
```

For :ref:`named and detached actors <actor-lifetimes>`, calling ``ray.kill`` on

> an actor handle will destroy the actor and allow the name to be reused.

Reviewer 2: ```suggestion
an actor handle destroys the actor and allow the name to be reused.
```

.. tip::
    

>     From Ray 2.0 onwards, you could view the actor exit details with :ref:`Ray's State API <state-api-overview-ref>``  to query the actors information, or go to the :ref:`Dashboard <observability-getting-started>`` Actor page to see more details. (These features all require installing ray with `pip install "ray[default]"`). 

Reviewer 2: ```suggestion
    From Ray 2.0 onwards, view the actor exit details by querying for information about the actor, with the :ref:`State API <state-api-overview-ref>``, or go to the Actors View in the :ref:`Dashboard <observability-getting-started>`` to see more details. (These features require installing Ray with `pip install "ray[default]"`). 
```

.. code-block:: bash
  # This API is only available when you download Ray via `pip install "ray[default]"`
  ray list actors --detail

.. code-block:: bash

>   # You should be able to view the death cause of an actor in `death_cause` as part of the actor state.

Reviewer: Here as well. Instead of having this code comment, just say, "you can verify actors are dead via this API."

---
-   actor_id: e8702085880657b355bf7ef001000000
    class_name: Actor
    state: DEAD
    job_id: '01000000'
    name: ''
    node_id: null
    pid: 0
    ray_namespace: dbab546b-7ce5-4cbb-96f1-d0f64588ae60
    serialized_runtime_env: '{}'
    required_resources: {}
    death_cause:
        actor_died_error_context: # <---- You could see the error message w.r.t why the actor exits. 

>             error_message: The actor is dead because it was killed by `ray.kill`.

Reviewer 2: ```suggestion
            error_message: The actor is dead because `ray.kill` killed it.
```

        can be waited on to wait for the actor to exit (calling ``ObjectRef::Get`` on it will
        throw a ``RayActorException``).

Note that this method of termination waits until any previously submitted
tasks finish executing and then exits the process gracefully with sys.exit.

    

> You could see the actor is dead as a result of the user's exit_actor() call:

Reviewer: Same as above

> Note that this method of termination will wait until any previously submitted

Reviewer 2: ```suggestion
Note that this method of termination waits until any previously submitted
```

tasks finish executing and then exit the process gracefully with sys.exit.

.. tip::
    

>     In this case you could see the actor death accordingly indicating users called exit_actor():

Reviewer 2: ```suggestion
    This response shows that the actor is dead as a result of the user's exit_actor() call:
```
